model/src/main/kotlin/PackageCuration.kt

```diff
@@ -64,7 +70,24 @@ data class PackageCuration(
      */
     private fun isApplicableIvyVersion(pkgId: Identifier) =
         runCatching {
-            Semver.coerce(pkgId.version).satisfies(id.version)
+            if (Semver.isValid(id.version)) {
+                // If the curation is for a valid semver, do not coerce the package version, as it also has to be valid.
+                return id.version == pkgId.version
+            }
+
+            if (id.version.isVersionRange()) {
+                // `Semver.satisfies(String)` requires a valid version range to work as expected, see:
+                // https://github.com/semver4j/semver4j/issues/132.
```

Comment:
This upstream issue has been closed meanwhile. Is it still sensible to mention it here?

Reply:
I think this can be removed now.
However, we still need to handle this case in our code.